src/editorAdapter.ts

```diff
@@ -1,5 +1,24 @@
-import { BookData } from "./interfaces";
-import { OPDSEntry } from "opds-feed-parser";
+import { BookData, LinkData } from "./interfaces";
+import { OPDSEntry, OPDSLink } from "opds-feed-parser";
+import {
+  PER_LIBRARY_SUPPRESS_REL,
+  PER_LIBRARY_UNSUPPRESS_REL,
+} from "./features/book/bookEditorSlice";
+
+/** Convert an OPDS link to a LinkData object. */
+const opdsLinkToLinkData = (link: OPDSLink | undefined): LinkData => {
+  if (!link) {
+    return link;
+  }
+  const {
+    href,
+    rel,
+    role = undefined,
+    title = undefined,
+    type = undefined,
+  } = link;
+  return { ...link, href, rel, title, type, role };
```

Comment:
`LinkData` is defined as: https://github.com/ThePalaceProject/circulation-admin/blob/9401ed129d51c24aad0981b308c98756e3e48969/src/interfaces.ts#L12-L15

Why do we need anything besides `href` and `rel` in the returned object?

Reply:
We might not. However, I included them because the objects that were returned previously were `OPDSLink`s, which do have those properties. And, if I'm remembering correctly, some of the test set/use those other properties.

Reply:
@ray-lee I went back and tested the following:
```typescript
const opdsLinkToLinkData = (link: OPDSLink | undefined): LinkData => {
  if (!link) {
    return link;
  }
  const { href, rel } = link;
  return { href, rel };
};
```

and got a failing test here:
```
  1 failing

  1) editorAdapter
       adapts valid OPDS entry:

      AssertionError: expected { Object (href, rel) } to deeply equal { Object (href, rel, ...) }
      + expected - actual

       {
         "href": "hide"
         "rel": "http://librarysimplified.org/terms/rel/hide"
      +  "role": "role"
      +  "title": "title"
      +  "type": "type"
       }
      
      at Context.<anonymous> (src/__tests__/editorAdapter-test.ts:140:38)
      at processImmediate (node:internal/timers:476:21)
```

But I could at least avoid explicitly mentioning `role`, `title`, and `type` here by changing the last two lines to
```typescript
  const { href, rel } = link;
  return { ...link, href, rel };
```

Do you have a preference among these options?
1. Leave the code as it is;
2. Use the last couple of lines, as above (return OPDSLink props besides `rel` and `href`); or
3. Drop the failing test, assuming that there is no functionality that depends on those other properties being present on the object, and return on `rel` and `href`.

If I leave it as is, I can update the `LinkData` interface to acknowledge the presence of these other properties.